packages/block-library/src/embed/edit.native.js

```diff
@@ -262,12 +258,23 @@ const EmbedEdit = ( props ) => {
 			) : (
 				<>
 					<EmbedControls
-						showEditButton={ preview && ! cannotEmbed }
 						themeSupportsResponsive={ themeSupportsResponsive }
 						blockSupportsResponsive={ responsive }
 						allowResponsive={ allowResponsive }
 						toggleResponsive={ toggleResponsive }
-						switchBackToURLInput={ () => setIsEditingURL( true ) }
+						bottomSheetLabel={ bottomSheetLabel }
+						url={ url }
+						onEmbedBottomSheetSubmit={ ( value ) => {
+							// The order of the following calls is important, we need to update the URL attribute before changing `isEditingURL`,
+							// otherwise the side-effect that potentially replaces the block when updating the local state won't use the new URL
+							// for creating the new block.
+							setAttributes( { url: value } );
+							setIsEditingURL( false );
+						} }
+						onEmbedBottomSheetClose={ () =>
+							setShowEmbedBottomSheet( false )
+						}
+						showEmbedBottomSheet={ showEmbedBottomSheet }
```

Comment:
These props are similar to how the `EmbedBottomSheet` is used within this Component. Let me know if the logic sharing approach here is fine or if there's a more optimal way to do this.

Reply:
Now that the visibility of `EmbedBottomSheet` component is controlled by `EmbedControls`, probably we could render `EmbedBottomSheet` within the `EmbedControls` component and only expose here a prop to notify when the URL is edited, wdyt?

Reply:
@fluiddot I think that approach is fine. Do we want to link picking behavior to be the same for both adding and editing a link? I assume that would make the experience more optimal. However, I noticed that when adding a link the embed bottom sheet would accept the input directly but when editing a link as done here, there would be a picker for that link because that is the approach that I see being utilized in other Block Settings. So the question is do we utilize inline editing instead of a Picker for both experiences? I think doing this would allow the experience to be optimal. ( At first, I had tried this but I think how the attributes were being set was possibly causing an issue so I didn't pursue it)

Now when I think about it, I agree with the case where if we utilize a prop to notify when a URL is being edited then, only at that time would we need to render the link editing component within the `BlockSettings` `PanelBody` otherwise we return the bottom sheet. i.e a function that does something similar to the link settings logic below. 

https://github.com/WordPress/gutenberg/blob/fbac29b09913c60f3004eb979542f851daf87f7a/packages/components/src/mobile/link-settings/index.native.js#L285

Reply:
> @fluiddot I think that approach is fine. Do we want to link picking behavior to be the same for both adding and editing a link? I assume that would make the experience more optimal. However, I noticed that when adding a link the embed bottom sheet would accept the input directly but when editing a link as done here, there would be a picker for that link because that is the approach that I see being utilized in other Block Settings. So the question is do we utilize inline editing instead of a Picker for both experiences? I think doing this would allow the experience to be optimal. ( At first, I had tried this but I think how the attributes were being set was possibly causing an issue so I didn't pursue it)

I think it makes sense to provide the same experience in both cases. I'd stick with the option where the user can edit directly the URL (as we're currently providing). The one that presents the picker can be confusing because it's related to linking the block to an URL, but in this case, we're editing the source of the embed.

> Now when I think about it, I agree with the case where if we utilize a prop to notify when a URL is being edited then, only at that time would we need to render the link editing component within the BlockSettings PanelBody otherwise we return the bottom sheet. i.e a function that does something similar to the link settings logic below.

Yep, exactly. I hope this doesn't introduce undesired side effects but looks like a promising way to go 👍 . Let me know if I can help anyhow in the implementation 🙇 .

Reply:
> Thanks for confirming 🙇🏾 My goal was to take a look at it after I am done WPiOS testing efforts and crash monitoring. If you would want to do the refactor and it doesn't take much effort from your side I am more than happy to review the changes afterward in an effort to keep things moving.

I'd like first to wrap up first the tests tasks but after that, I'll be more than happy to help with this implementation. I'll let you know if I start tackling it 👍 .